pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -253,13 +253,19 @@ func (ks *scaler) applyScale(ctx context.Context, pa *pav1alpha1.PodAutoscaler,
 // Scale attempts to scale the given PA's target reference to the desired scale.
 func (ks *scaler) Scale(ctx context.Context, pa *pav1alpha1.PodAutoscaler, desiredScale int32) (int32, error) {
 	logger := logging.FromContext(ctx)
+	autoscalerConfig := config.FromContext(ctx).Autoscaler
 
 	if desiredScale < 0 {
-		logger.Debug("Metrics are not yet being collected.")
-		return desiredScale, nil
+		// check is activating timeout
+		if pa.Status.IsActivatingTimeout(autoscalerConfig.ActivatingTimeout) {
```

Comment:
I don't think that this is correct. It should not really matter in the negative case at all.